crates/cheatcodes/src/inspector.rs

```diff
@@ -392,6 +412,123 @@ impl<DB: DatabaseExt> Inspector<DB> for Cheatcodes {
             }
         }
 
+        // Record account access via SELFDESTRUCT if `recordAccountAccesses` has been called
+        if let Some(account_accesses) = &mut self.recorded_account_diffs {
+            if interpreter.current_opcode() == opcode::SELFDESTRUCT {
+                let target = try_or_continue!(interpreter.stack().peek(0));
+                // load balance of this account
+                let value: U256;
+                if let Ok((account, _)) =
+                    data.journaled_state.load_account(interpreter.contract().address, data.db)
+                {
+                    value = account.info.balance;
+                } else {
+                    value = U256::ZERO;
+                }
+                // previous balance of the target account
+                let old_balance: U256;
+                // get initialized status of target account
+                let initialized: bool;
+                if let Ok((account, _)) =
+                    data.journaled_state.load_account(Address::from(U160::from(target)), data.db)
+                {
+                    initialized = account.info.exists();
+                    old_balance = account.info.balance;
+                } else {
+                    initialized = false;
+                    old_balance = U256::ZERO;
+                }
+
+                // register access for the target account
+                let access = crate::Vm::AccountAccess {
+                    chainInfo: crate::Vm::ChainInfo {
+                        forkId: data.db.active_fork_id().unwrap_or_default(),
+                        chainId: U256::from(data.env.cfg.chain_id),
+                    },
+                    accessor: interpreter.contract().address,
+                    account: Address::from(U160::from(target)),
```

Comment:
This doesn't work as you'd expect. U160::from panics on overflow, you want `Address::from_word(B256::from(target))`. Also this can be moved at the start of the block to remove duplication in `load_account`.